### PR TITLE
Closes issue #203 - avoid calling sizeInBytes if it's not expected to exist

### PR DESCRIPTION
**Pull request by matthpeterson**

### CODE REVIEW

**Comment by matthpeterson** [comment]
@billoley are you ok with this MR?  Jstack showed a lot of thrown exceptions on the now removed code and so I think it is best to remove it.

File: client/src/main/java/timely/model/ObjectSizeOf.java [hunk 1]
```diff
@@ -87,15 +86,8 @@ public static long getObjectSize(Object o, Set<ObjectInstance> visited, Stack<Ob
...
-                            try {
-                                if (o instanceof ObjectSizeOf) {
-                                    size = ((ObjectSizeOf) o).sizeInBytes();
-                                } else {
-                                    Method sizeInBytes = o.getClass().getMethod("sizeInBytes", (Class<?>[]) null);
-                                    size = (Long) sizeInBytes.invoke(o);
-                                }
-                            } catch (Throwable t) {
-                                // ok, lets do this the hard way...
+                            if (o instanceof ObjectSizeOf) {
```
Comment (jschmidt10) [inline client/src/main/java/timely/model/ObjectSizeOf.java:89]: log warning if it's another object so we can hunt it down?

Comment (jschmidt10) [inline client/src/main/java/timely/model/ObjectSizeOf.java:89]: It'd be cool to create a wrapper for HashMap<Key,Value> which implements ObjectSizeOf. I know we might not do it soon but is it worth opening an issue for that?

**Comment by dlmarion** [comment]
Another approach would be to fix the code such that it does not throw the NoSuchMethodException. You could check if the Method exists in the class' declared methods, then invoke it if it exists. As an optimization, you could keep a list of classes that had previous successful checks.

File: client/src/main/java/timely/model/ObjectSizeOf.java (hunk 1 above)
Comment (matthpeterson) [reply client/src/main/java/timely/model/ObjectSizeOf.java:89]: It seems that could be worthwhile for a separate ticket.  I wanted to get a low effort performance improvement with this MR.

**Comment by matthpeterson** [comment]
@dlmarion for some reason I couldn't reply directly to your comment. I considered that option as well and ended up going this route because it seemed unlikely that an object would have implemented sizeInBytes unless it implements that interface and I wanted to avoid the cost of getting the methods and looking through them.

**Comment by billoley** [comment]
I don't see _anything_ that implements the ObjectSizeOf method.  I think this code was copied either from datawave or accumulo and used to estimate when to return from the DownsampleIterator to avoid blowing up the tserver.  It would be a good idea as a followon to implement that interface on Downsample and Tag as a start.  Might lead to better performance in all queries.  Good catch!

I agree with your logic (class should implement the interface if it wants that call) and the change.

**Comment by billoley** [comment]
Correction - DownsampleIteratorTest tests this approach (implementing ObjectSizeOf) , but nothing else yet.

**Comment by brianloss** [comment]
> 
> 
> > I don't see _anything_ that implements the ObjectSizeOf method. I think this code was copied either from datawave or accumulo and used to estimate when to return from the DownsampleIterator to avoid blowing up the tserver. It would be a good idea as a followon to implement that interface on Downsample and Tag as a start. Might lead to better performance in all queries. Good catch!
> > I agree with your logic (class should implement the interface if it wants that call) and the change.
> 
> FYI, [BaseType](https://github.com/NationalSecurityAgency/datawave-type-utils/blob/master/src/main/java/datawave/data/type/BaseType.java) implements the interface and a lot of things extend that.

Hah, I was looking at datawave. Never mind. :)

**Comment by matthpeterson** [comment]
Thanks for the reviews.  I don't have merge privileges.

**Pull request merged**